File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int candidate = 0, count = 0;
        int n = nums.length;
        int i = 0;
        while(i < n){
            if(count == 0){
                candidate = nums[i];
            }
            if(nums[i] == candidate){
                count++;
            }
            else{
                count--;
            }
            i++;
        }

        return candidate;
    }
}

File: 169-majority-element/majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        count, candidate = 0, 0
        n = len(nums)
        for i in range(n):
            if count == 0:
                candidate = nums[i]
                count += 1
            elif nums[i] == candidate:
                count += 1
            else:
                count -= 1
        
        return candidate
